pythonmods/vlcplayer/__init__.py

```diff
@@ -37,10 +38,28 @@ def __init__(self):
         self.sayplayer = self.instance.media_player_new()
         self.list_player =  self.instance.media_list_player_new()
         self.list_player.set_media_player(self.player)
+        self.sc_api = SoundcloudAPI()
 
     def playytb(self, vid):
         self.play(vid2youtubeMRL(vid))
 
+    def playLocalFiles(self,url):
+        # Method to play local mp3 files
+        media_list = self.instance.media_list_new(url)
+        self.list_player.set_media_list(media_list)
+        self.list_player.play()
+        self.softvolume(100, self.player)
+
+    def playscloud(self, url):
+        #Url of the format: https://soundcloud.com/aries_ix/sayonara
+        track = self.sc_api.resolve(url)
+        assert type(track) is Track
+        filename = os.path.join(os.path.dirname(os.path.realpath(__file__)),"download.mp3")
+        with open(filename, "wb+") as fp:
+            track.write_mp3_to(fp)
+        # subprocess.call(["play","download.mp3"])
+        self.playLocalFiles(filename)
```

Comment:
Why don't you use `self.play(filename)`? It should work as far as I remember. Then the duplication of `playLocalFiles` can be removed